aten/src/ATen/native/TensorIterator.h

```diff
@@ -20,49 +22,61 @@
 //      .build()
 //
 // [MyKernel.cpp / MyKernel.cu]
-//   cpu_binary_kernel(iter, [](float a, float b) {
+//   binary_kernel(iter, [](float a, float b) {
 //     return a + b;
 //   });
 //
 //   gpu_binary_kernel(iter, []GPU_LAMBDA(float a, float b) -> float {
 //     return a + b;
 //   });
 //
-// This inspired by NumPy's Array Iterator API (NpyIter).
+// Note [Result type computation]
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// TensorIterator handles limited mixed-type operations. The result type is
+// computed using promoteTypes on the types of the operands with the following
```

Comment:
I'd try to avoid saying types in ATen ("on the types)"...you mean scalar types here, correct?